core/src/main/java/bisq/core/btc/wallet/BsqCoinSelector.java

```diff
@@ -32,18 +34,34 @@
  */
 @Slf4j
 public class BsqCoinSelector extends BisqDefaultCoinSelector {
-    private DaoStateService daoStateService;
+    private final DaoStateService daoStateService;
+    private final UnconfirmedBsqChangeOutputListService unconfirmedBsqChangeOutputListService;
 
     @Inject
-    public BsqCoinSelector(DaoStateService daoStateService) {
+    public BsqCoinSelector(DaoStateService daoStateService, UnconfirmedBsqChangeOutputListService unconfirmedBsqChangeOutputListService) {
         super(true);
         this.daoStateService = daoStateService;
+        this.unconfirmedBsqChangeOutputListService = unconfirmedBsqChangeOutputListService;
     }
 
     @Override
     protected boolean isTxOutputSpendable(TransactionOutput output) {
         // output.getParentTransaction() cannot be null as it is checked in calling method
-        return output.getParentTransaction() != null &&
-                daoStateService.isTxOutputSpendable(new TxOutputKey(output.getParentTransaction().getHashAsString(), output.getIndex()));
+        Transaction parentTransaction = output.getParentTransaction();
+        if (parentTransaction == null)
+            return false;
+
+        // If it is a normal confirmed BSQ output we use the default lookup at the daoState
+        if (daoStateService.isTxOutputSpendable(new TxOutputKey(parentTransaction.getHashAsString(), output.getIndex())))
+            return true;
+
+        // It might be that it is an unconfirmed change output which we allow to be used for spending without requiring a confirmation.
+        // We check if we have the output in the dao state, if so we have a confirmed but unspendable output (e.g. confiscated).
+        if (daoStateService.getTxOutput(new TxOutputKey(parentTransaction.getHashAsString(), output.getIndex())).isPresent())
+            return false;
+
+        // Only if its not existing yet in the dao state (unconfirmed) we use our unconfirmedBsqChangeOutputList to
+        // check if it is an own change output.
+        return unconfirmedBsqChangeOutputListService.hasTransactionOutput(output);
```

Comment:
Don't we need to check that it's also unspent or is the unconfirmed list only for unspent txs?

Reply:
As we remove entries for all connected outputs of any inputs we should not get that problem (e.g. user spend utxo from previously unconfirmed tx). If we would miss one it would throw an error on the BitcoinJ level if we try to use an input which is already spent. But I will have a look to add an additional check.